homework/hackyourtemperature/__tests__/app.test.js

```diff
@@ -0,0 +1,49 @@
+import app from '../app.js';
+import supertest from 'supertest';
+
+const request = supertest(app);
+
+describe('POST /weather', () => {
+    describe('given a city name', () => {
+        const body = {
+            cityName: 'Amsterdam'
+        };
+        it('should respond with a 200 status code', async () => {
+            const response = await request.post('/weather').send(body);
+            expect(response.statusCode).toBe(200);
+        });
+        it('should specify json in the content type header', async () => {
+            const response = await request.post('/weather').send(body);
+            expect(response.headers['content-type']).toEqual(
+                expect.stringContaining('json')
+            );
+        });
+    });
+
+    describe('when the city name is', () => {
+        it('wrong, it should respond with a status code 404', async () => {
```

Comment:
I wouldn't write my `it` statements like this. I would do

```suggestion
    describe('when the city name is invalid', () => {
        it.each`
          cityName | statusCode
          ${"Ansterdam"} | ${404}
          ${"dhfdhfdj"} | ${404}
          ${""} | ${400}
        `('should respond with a status code $statusCode', async (cityName, statusCode) => {
        ... do test
```

A bit advanced testing but it will make your tests shorter and faster 💪 

Even without this, try to not include the word `it` in your `it`-string. It should just read out like a normal sentence